File: src/main/java/io/metjka/vortex/ui/blocks/ConstantBlock.java
package io.metjka.vortex.ui.blocks;

import com.google.common.collect.ImmutableList;
import io.metjka.vortex.ui.ToplevelPane;
import io.metjka.vortex.ui.connections.OutputAnchor;
import javafx.fxml.FXML;
import javafx.scene.control.TextField;
import javafx.scene.layout.Pane;

import java.util.List;
import java.util.Optional;

public class ConstantBlock extends ValueBlock<Integer> {

    private final OutputAnchor outputAnchor = new OutputAnchor(this, 0, Type.NUMBER);
    @FXML
    private TextField textField;

    @FXML
    private Pane outputSpace;

    public ConstantBlock(ToplevelPane pane) {
        super(pane, "ConstantBlock");

        outputSpace.getChildren().add(0, outputAnchor);

        textField.setText("0");
        value1 = 0;

        textField.textProperty().addListener((observable, oldValue, newValue) -> {
            if (!"".equals(newValue)) {
                value1 = Integer.parseInt(newValue);
                sendUpdateDownSteam();
            } else {
                value1 = 0;
                sendUpdateDownSteam();
            }
        });

    }

    @Override
    public void update() {
        outputAnchor.invalidateVisualState();
    }

    @Override
    public List<OutputAnchor> getAllOutputs() {
        return ImmutableList.of(outputAnchor);
    }

    @Override
    public List<InputAnchor> getAllInputs() {
        return ImmutableList.of();
    }

    @Override
    public Optional<Block> getNewCopy() {
        return null;
    }

    @Override
    Integer getValue(int position) {
        return value1;
    }
}

File: src/main/java/io/metjka/vortex/ui/blocks/ConstantBlock.kt
package io.metjka.vortex.ui.blocks

import io.metjka.vortex.ui.TopLevelPane
import io.metjka.vortex.ui.connections.InputDot
import io.metjka.vortex.ui.connections.OutputDot
import javafx.fxml.FXML
import javafx.scene.layout.Pane

class ConstantBlock(topLevelPane: TopLevelPane) : NodeBlock(topLevelPane, ConstantBlock::class.simpleName) {

    @FXML
    lateinit var outputSpace: Pane

    val outputDot = OutputDot<Number>(this)

    init {
        outputSpace.children?.add(0, outputDot)
    }

    override fun update() {

    }

    override fun getAllInputs(): List<InputDot<*>> {
        return emptyList()
    }

    override fun getAllOutputs(): List<OutputDot<*>> {
        return listOf(outputDot)
    }

    override fun getNewCopy(): NodeBlock {
        return this
    }

}